Segfault with incorrect registration token

```
$ buildkite-agent start --token Foo

  _           _ _     _ _    _ _                                _
 | |         (_) |   | | |  (_) |                              | |
 | |__  _   _ _| | __| | | ___| |_ ___    __ _  __ _  ___ _ __ | |_
 | '_ \| | | | | |/ _` | |/ / | __/ _ \  / _` |/ _` |/ _ \ '_ \| __|
 | |_) | |_| | | | (_| |   <| | ||  __/ | (_| | (_| |  __/ | | | |_
 |_.__/ \__,_|_|_|\__,_|_|\_\_|\__\___|  \__,_|\__, |\___|_| |_|\__|
                                                __/ |
 http://buildkite.com/agent                    |___/

2019-07-17 23:54:16 NOTICE Starting buildkite-agent v3.13.1 with PID: 28253
2019-07-17 23:54:16 NOTICE The agent source code can be found here: https://github.com/buildkite/agent
2019-07-17 23:54:16 NOTICE For questions and support, email us at: hello@buildkite.com
2019-07-17 23:54:16 INFO   Configuration loaded path=/usr/local/etc/buildkite-agent/buildkite-agent.cfg
2019-07-17 23:54:16 INFO   Registering agent with Buildkite...
2019-07-17 23:54:17 WARN   Buildkite rejected the registration (POST https://agent.buildkite.com/v3/register: 401 The token you provided "Foo" is not a valid organization agent registration token.)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x1a11442]

goroutine 1 [running]:
github.com/buildkite/agent/agent.Register(0x2031c80, 0xc000242640, 0x2037b40, 0xc000059380, 0xc000023146, 0xc, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/buildkite/agent/agent/register.go:61 +0x362
github.com/buildkite/agent/clicommand.glob..func1(0xc00024c000)
	/go/src/github.com/buildkite/agent/clicommand/agent_start.go:623 +0x1265
github.com/buildkite/agent/vendor/github.com/urfave/cli.HandleAction(0x1b871a0, 0x1e868d8, 0xc00024c000, 0x0, 0xc000059260)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:503 +0x7c
github.com/buildkite/agent/vendor/github.com/urfave/cli.Command.Run(0x1e45303, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1e58903, 0x18, 0x0, ...)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/command.go:165 +0x487
github.com/buildkite/agent/vendor/github.com/urfave/cli.(*App).Run(0xc000226000, 0xc00001e100, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:259 +0x6e3
main.main()
	/go/src/github.com/buildkite/agent/main.go:114 +0x52f
$ echo $?
2
```

I just got the same :-(

```
$ buildkite-agent start --token=blah --endpoint=http://agent.buildkite.localhost/v3

  _           _ _     _ _    _ _                                _
 | |         (_) |   | | |  (_) |                              | |
 | |__  _   _ _| | __| | | ___| |_ ___    __ _  __ _  ___ _ __ | |_
 | '_ \| | | | | |/ _` | |/ / | __/ _ \  / _` |/ _` |/ _ \ '_ \| __|
 | |_) | |_| | | | (_| |   <| | ||  __/ | (_| | (_| |  __/ | | | |_
 |_.__/ \__,_|_|_|\__,_|_|\_\_|\__\___|  \__,_|\__, |\___|_| |_|\__|
                                                __/ |
 http://buildkite.com/agent                    |___/

2019-07-19 16:12:13 NOTICE Starting buildkite-agent v3.13.1 with PID: 63801
2019-07-19 16:12:13 NOTICE The agent source code can be found here: https://github.com/buildkite/agent
2019-07-19 16:12:13 NOTICE For questions and support, email us at: hello@buildkite.com
2019-07-19 16:12:13 INFO   Configuration loaded path=$HOME/.buildkite-agent/buildkite-agent.cfg
2019-07-19 16:12:13 INFO   Registering agent with Buildkite...
2019-07-19 16:12:13 WARN   Buildkite rejected the registration (POST http://agent.buildkite.localhost/v3/register: 401 The token you provided "blah" is not a valid organization agent registration token.)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x58 pc=0x1a11442]

goroutine 1 [running]:
github.com/buildkite/agent/agent.Register(0x2031c80, 0xc000282500, 0x2037b40, 0xc00009e1e0, 0xc0000b8fa6, 0xc, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/buildkite/agent/agent/register.go:61 +0x362
github.com/buildkite/agent/clicommand.glob..func1(0xc00028e000)
	/go/src/github.com/buildkite/agent/clicommand/agent_start.go:623 +0x1265
github.com/buildkite/agent/vendor/github.com/urfave/cli.HandleAction(0x1b871a0, 0x1e868d8, 0xc00028e000, 0x0, 0xc0000a3200)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:503 +0x7c
github.com/buildkite/agent/vendor/github.com/urfave/cli.Command.Run(0x1e45303, 0x5, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1e58903, 0x18, 0x0, ...)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/command.go:165 +0x487
github.com/buildkite/agent/vendor/github.com/urfave/cli.(*App).Run(0xc000266000, 0xc0000a4000, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/buildkite/agent/vendor/github.com/urfave/cli/app.go:259 +0x6e3
main.main()
	/go/src/github.com/buildkite/agent/main.go:114 +0x52f
```

--- a/agent/register.go
+++ b/agent/register.go
@@ -56,7 +56,7 @@ func Register(l logger.Logger, ac APIClient, req api.AgentRegisterRequest) (*api
 
 	// Try to register, retrying every 10 seconds for a maximum of 30 attempts (5 minutes)
 	err = retry.Do(register, &retry.Config{Maximum: 30, Interval: 10 * time.Second})
-	if err != nil {
+	if err == nil {
 		l.Info("Successfully registered agent \"%s\" with tags [%s]", registered.Name,
 			strings.Join(registered.Tags, ", "))
 

